cmd/thanos/query.go

```diff
@@ -48,6 +48,7 @@ func registerQuery(m map[string]setupFunc, app *kingpin.Application, name string
 	cert := cmd.Flag("grpc-client-tls-cert", "TLS Certificates to use to identify this client to the server").Default("").String()
 	key := cmd.Flag("grpc-client-tls-key", "TLS Key for the client's certificate").Default("").String()
 	caCert := cmd.Flag("grpc-client-tls-ca", "TLS CA Certificates to use to verify gRPC servers").Default("").String()
+	serverName := cmd.Flag("grpc-client-server-name", "Server name to verify the hostname on the returned gRPC certificates").Default("").String()
```

Comment:
Can we add `https://tools.ietf.org/html/rfc4366#section-3.1` link somewhere here?